web/src/main/java/org/wso2/testgrid/web/api/TestPlanService.java

```diff
@@ -394,4 +401,71 @@ private TestExecutionSummary getTestExecutionSummary(TestPlan testPlan) {
         }
         return new TestExecutionSummary(scenarioSummaries, scenarioTestCaseEntries);
     }
+
+    /**
+     * Returns the result (archive file) to the given scenario of a test plan.
+     *
+     * @param id test plan id
+     * @param scenarioDir scenario directory name
+     * @return archived file of the results
+     */
+    @GET
+    @Path("/result/{id}/{scenarioDir}")
```

Comment:
Shall we say 'test-plan-id' to make it more meaningful? Then we don't have to look at the javadoc.. :)

Reply:
What is the scenarioDir?? If this is a jargon we are going to use, we should document this somewhere.